livelog/livelog.go

```diff
@@ -97,13 +103,25 @@ func (l *LiveLog) connectInputStream() error {
 	}
 	client := new(http.Client)
 	go func() {
-		// TODO: this is a HORRENDOUS hack - need to fix this Basically we need
-		// to wait until put port is opened which is some time after the
+		// We need to wait until put port is opened which is some time after the
 		// livelog process has started...
-		time.Sleep(500 * time.Millisecond)
+		waitForPortToBeActive(60022)
 		// since we waited so long, maybe livelog service isn't running now, so
 		// ignore any error and response we get back...
 		client.Do(req)
 	}()
 	return nil
 }
+
+func waitForPortToBeActive(port int) {
+	deadline := time.Now().Add(60 * time.Second)
+	for time.Now().Before(deadline) {
+		conn, err := net.DialTimeout("tcp", "localhost:"+strconv.Itoa(port), 60*time.Second)
```

Comment:
Ah thanks, i didn't know that.  Did you discover that from documentation or by looking at the code? I think I might have missed something.